keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,416 @@
+    - [Topology information in the NodeResourceTopologyMatch plugin](#topology-information-in-the-noderesourcetopologymatch-plugin)
+    - [Description of the Algorithm](#description-of-the-algorithm)
+  - [Accessing NodeResourceTopology CRD](#accessing-noderesourcetopology-crd)
+- [Use cases](#use-cases)
+- [Known limitations](#known-limitations)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature enablement and rollback](#feature-enablement-and-rollback)
+- [Implementation history](#implementation-history)
+<!-- /toc -->
+
+# Summary
+
+Kubernetes clusters composed of nodes with complex hardware topology are becoming more prevalent.
+[Topology Manager](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/) was
+introduced in kubernetes as part of kubelet in order to extract the best performance out of
+these high performance hybrid systems. It performs optimizations related to resource allocation
+in order to make it more likely for a given pod to perform optimally. In scenarios where
+Topology Manager is unable to align topology of requested resources based on the selected
+Topology Manager policy, the pod is rejected with Topology Affinity Error.
+[This](https://github.com/kubernetes/kubernetes/issues/84869) kubernetes issue provides
+further context on how runaway pods are created because the scheduler is topology-unaware.
+
+In order to address this issue, scheduler needs to choose a node considering resource availability
+along with underlying resource topology and Topology Manager policy on the worker node.
+
+This document describes behaviour of the Kubernetes Scheduler which takes worker node topology into account.
+
+# Motivation
+
+After Topology Manager was introduced, the problem of launching pod in the cluster where worker
+nodes have different NUMA topology and different amount of resources in that topology became
+actual. Pod could be scheduled on the node where total amount of resources are enough, but
+resource distribution could not satisfy the appropriate Topology policy. In this case the pod
+failed to start. Much better behaviour for scheduler would be to select appropriate node where
+kubelet admit handlers may pass.
+
+In order to enable topology aware scheduling in Kubernetes, resource topology information of the
+nodes in the cluster needs to be exposed to the scheduler so that it can use it to make a more
+informed scheduling decision. This KEP describes how it would be implemented.
+
+## Goals
+
+- Make scheduling process more precise when we have NUMA topology on the
+worker node.
+- Enhance the node object to capture topology information which can be referred to 
+by the scheduler.
+
+## Non-Goals
+
+- Change the PodSpec to allow requesting a specific node topology manager policy
+- This Proposal requires exposing NUMA topology information. This KEP doesn't
```

Comment:
Kubelet does know about the allocated resources and its topology but we are trying expose informatio pertaining to the available resource along with the granularity of NUMA nodes which is not exposed in Kubelet.

Reply:
great, I think we need to highlight that this will not require new APIs with the OS and whatnot, stress that this information is already known by kubelet. Also, clarify which operating systems are currently supported.